tests/src/Kernel/TypedData/DataFetcherTest.php

```diff
@@ -70,4 +82,85 @@ public function testFetchingByBasicPropertyPath() {
     );
   }
 
-}
\ No newline at end of file
+  /**
+   * @cover fetchBySubPaths
+   */
+  public function testFetchingByBasicSubPath() {
+    $this->assertEquals(
+      $this->node->title->value,
+      $this->typedDataManager->getDataFetcher()
+      ->fetchBySubPaths($this->node->getTypedData(), array('title', '0', 'value'))
+      ->getValue()
+    );
+  }
+
+  /**
+   * @cover fetchByPropertyPath
+   */
+  public function testFetchingEntityReference() {
+    $user = $this->entityTypeManager->getStorage('user')
+      ->create([
+        'name' => 'test',
+        'type' => 'user',
+      ]);
+    $this->node->uid->target_id = $user->id();
+
+    $fetched_user = $this->typedDataManager->getDataFetcher()
+      ->fetchByPropertyPath($this->node->getTypedData(), 'uid.entity.value')
+      ->getValue();
+    $this->assertEquals(TRUE, $fetched_user instanceof EntityAdapter);
+  }
+
+  /**
+   * @cover fetchByPropertyPath
+   */
+  public function testFetchingEntityReferenceAtPosition0() {
```

Comment:
position tests should use a multi-value field, that does not have to be a entity reference though. do not mix up things and clearly document what you are testing